common/lib/client/auth.ts

```diff
@@ -863,7 +864,7 @@ class Auth {
 
 
 	/* User-set: check types, '*' is disallowed, throw any errors */
-	_userSetClientId(clientId: string | null) {
+	_userSetClientId(clientId: string | undefined) {
```

Comment:
What's the purpose behind this change? I'm sure it was necessary but it does concern me that two lines below we have "clientId must be either a string or null"

Reply:
`_userSetClientId` takes `authOptions.clientId` which is an optional string, typescript treats it as if it were typed as `string | undefined` rather than `string | null`. The only option here was to either change the signature of this method, or change the public typings.